mathesar/rpc/roles.py

```diff
@@ -116,3 +116,26 @@ def add(
     with connect(database_id, user) as conn:
         role = create_role(rolename, password, login, conn)
     return RoleInfo.from_dict(role)
+
+
+@rpc_method(name="roles.get_current_role")
+@http_basic_auth_login_required
+@handle_rpc_exceptions
+def get_current_role(*, database_id: int, **kwargs) -> dict:
+    """
+    Get information about the current role and all the parent role(s) it inherits from.
```

Comment:
If we make the `'MEMBER'` -> `'USAGE'` switch, we might have to modify this to be a bit more precise; something like: "...and all the parent roles it can use in Mathesar"